ulimit_unix_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2019, Daniel Martí <mvdan@mvdan.cc>
+// See LICENSE for licensing information
+
+// TODO: replace with the unix build tag once we require Go 1.19 or later
+//go:build linux
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"syscall"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+	exec "golang.org/x/sys/execabs"
+)
+
+func init() {
+	// Here rather than in TestMain, to reuse the unix build tag.
+	if limit := os.Getenv("TEST_WITH_FILE_LIMIT"); limit != "" {
+		n, err := strconv.ParseUint(limit, 10, 64)
+		if err != nil {
+			panic(err)
+		}
+		rlimit := syscall.Rlimit{Cur: n, Max: n}
```

Comment:
Generally you'll want x/sys/unix for weird syscalls because it'll get updates if things ever change on various UNIXes.

Reply:
I considered this but guessed it was harmless given it's a test. I guess it's equally harmless to use x/sys/unix, and we already pull x/sys for other purposes. Done